crypto library does not load on osx and windows jdk11

enasser opened the pull request:
The issue with mac OSX is that libjncrypto.dylib is linked to
/usr/local/lib/libcrypto.1.1.dylib instead of linking with rpath
i.e @rpath/libcrypto.1.1.dylib.
This is because libcrypto.1.1.dylib by default is built with
/usr/local/lib as the path instead of rpath.
To fix the issue, updated the path entry of libcrypto.1.1.dylib
with @rpath using install_name_tool.

Signed-off-by: Nasser Ebrahim <enasser@in.ibm.com>

enasser commented:
@pshipton Please review the fix and trigger necessary tests. I have built and tested the mac fix locally.  I have included the fix for windows as well. However, I could not build it locally due to machine issues.

pshipton commented:
@enasser please fix the copyright
@keithc-ca please take a look

pshipton commented:
Once approved, the commit will need to be cherry picked and a PR created against the openj9-0.12 release branch.

keithc-ca commented on closed/autoconf/custom-hook.m4:
```diff
@@ -449,35 +449,72 @@ AC_DEFUN([CONFIGURE_OPENSSL],
+              # On Mac OSX, create local copy of the crypto library to update @rpath
+              # as the default is /usr/local/lib.
+              if test "x$OPENJDK_BUILD_OS" = "xmacosx" ; then
+                MKDIR = "mkdir"
```
This won't work: you need write things `bash` will understand, not content suitable for a makefile.
Please also avoid tabs in this file (for consistency).

enasser replied:
Thank you @keithc-ca . I have deleted the definition MKDIR as it's already defined.

keithc-ca commented on closed/autoconf/custom-hook.m4:
```diff
@@ -449,35 +449,72 @@ AC_DEFUN([CONFIGURE_OPENSSL],
+              if test "x$OPENJDK_BUILD_OS" = "xmacosx" ; then
+                MKDIR = "mkdir"
+                LOCAL_CRYPTO = "$TOPDIR/openssl"
+		$(MKDIR) $(LOCAL_CRYPTO)
```
This needs to be `mkdir -p "$LOCAL_CRYPTO"` in case the directory already exists.

enasser replied:
Done.

pshipton commented:
@enasser you resolved the comments but didn't push any new change.

pshipton commented:
The testing passed for both osx and win.

enasser commented:
@keithc-ca Thank you for your review comments. I have addressed your review comments.

@pshipton The review comments are addressed, built and tested on Mac, squash and push the changes. Please review and merge.

pshipton commented:
jenkins compile zlinux,win,osx

pshipton commented:
@enasser thanks Nasser. Keith will look again, but meanwhile I have started some testing to confirm.
Can you please look at similar changes for osx on jdk8, which is the only other thing remaining for the openj9 0.12 release.

enasser commented:
Thank you @pshipton. I have already made the changes for openjdk8 https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/244. I could not build it myself on mac locally as it has dependency on older mac packages. However, am not expecting any failures as the changes are similar. Please trigger the build for openjdk8 as well.
